### ISSUE TITLE
Timeout in sync from rest api

### ISSUE THREAD

**Issue by HiddenTesla** [argoproj/argo-cd#19815]
# Summary
How to set timeout in app sync?
This seems doable from CLI, as the `argocd app sync --help` says:
```
      --timeout uint                          Time out after this many seconds
```
However, swagger-ui `/swagger-ui#operation/ApplicationService_Sync` does not mention it. 
I have gone through all expandable items but do not find timeout.
![image](https://github.com/user-attachments/assets/96199b7d-aaed-4436-8143-1fcf14eb22f3)

# Motivation
To void sync takes too long.

# Proposal
This may be a question rather than an enhancement. But I do not find where to ask questions.

**Comment by HiddenTesla** [comment]
Hi do we have a quick anwser if this is available now or needs implementing?

**Comment by HiddenTesla** [comment]
> you need to pass an argument to the `-timeout` flag, something like
> 
> ```
> > argocd app sync app1 --timeout 2
> 
> TIMESTAMP                  GROUP        KIND   NAMESPACE                  NAME    STATUS   HEALTH        HOOK  MESSAGE
> 2024-09-10T17:25:11+05:30   apps  Deployment        app1                  app1    Synced  Healthy              
> 
> This is the state of the app after `wait` timed out:
> ```

Yes `--timeout` works for cli, but I wonder if there is an equivalnent option for its rest api.

**Comment by eugene70** [comment]
@HiddenTesla,

`cmd/argocd/commands/app.go`-`func NewApplicationSyncCommand`
```
_, err = appIf.Sync(ctx, &syncReq)
errors.CheckError(err)

if !async {
	app, opState, err := waitOnApplicationStatus(ctx, acdClient, appQualifiedName, timeout, watchOpts{operation: true}, selectedResources, output)
```
Calling the API from the CLI appears to be asynchronous behaviour.
It's just that after calling the API from the CLI, the CLI waits for a TIMEOUT for the state to change; there is no TIMEOUT because the API itself is asynchronous.
If you plan to call the API directly, it would make sense to implement TIMEOUT in the CLIENT.
